beacon-chain/db/filesystem/pruner.go

```diff
@@ -0,0 +1,273 @@
+package filesystem
+
+import (
+	"encoding/binary"
+	"io"
+	"path"
+	"path/filepath"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/pkg/errors"
+	fieldparams "github.com/prysmaticlabs/prysm/v4/config/fieldparams"
+	"github.com/prysmaticlabs/prysm/v4/config/params"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/primitives"
+	"github.com/prysmaticlabs/prysm/v4/time/slots"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+)
+
+const retentionBuffer primitives.Epoch = 2
+
+var (
+	errPruningFailures = errors.New("blobs could not be pruned for some roots")
+)
+
+type blobPruner struct {
+	sync.Mutex
+	prunedBefore atomic.Uint64
+	windowSize   primitives.Slot
+	slotMap      *slotForRoot
+	fs           afero.Fs
+}
+
+func newblobPruner(fs afero.Fs, retain primitives.Epoch) (*blobPruner, error) {
```

Comment:
```suggestion
func newBlobPruner(fs afero.Fs, retain primitives.Epoch) (*blobPruner, error) {
```

Reply:
ty, made this change by hand since callsites also need to change.